File: Backend/src/main/java/net/ethlny/discordhetic/discord_backend_hetic/models/Role.java
package net.ethlny.discordhetic.discord_backend_hetic.models;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.io.Serializable;
import java.util.UUID;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "roles")
public class Role implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "uuid", updatable = false)
    private UUID id;

    @Column(nullable = false, unique = true)
    private String name;

    // Optional: add a description field if needed
    private String description;

    public Role() {
    }

    public Role(String name) {
        this.name = name;
    }
}

File: Frontend/Models/Role.cs
using System;

namespace DiscordFrontEnd.Models {
    public class Role {
        public Guid Id {
            get; set;
        }
        public string Name {
            get; set;
        }
        public string Description {
            get; set;
        }

        public Role() {
        }

        public Role(string name) {
            Name = name;
        }
    }
}
